packages/core/phase3/lib/pncUpdateRequestGenerators/remandGenerator/remandGenerator.test.ts

```diff
@@ -0,0 +1,217 @@
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.BailConditions = [
+    "This is a dummy bail condition."
+  ]
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.ForceOwner = {
+    TopLevelCode: "A",
+    SecondLevelCode: "02",
+    ThirdLevelCode: "BJ",
+    BottomLevelCode: "01",
+    OrganisationUnitCode: "A02BJ01"
+  }
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing.CourtType = "MCA"
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing.CourtHouseName = "Magistrates' Courts London Croydon"
+
+  return pncUpdateDataset
+}
+
+const operation = {
+  code: "NEWREM",
+  data: {
+    nextHearingDate: new Date("2024-12-11T10:11:12.000Z"),
+    nextHearingLocation: {
+      TopLevelCode: "B",
+      SecondLevelCode: "00",
+      ThirdLevelCode: "00",
+      BottomLevelCode: "00",
+      OrganisationUnitCode: "B000000"
+    }
+  }
+} as Operation<PncOperation.REMAND>
+
+describe("remandGenerator", () => {
+  it("should generate remand request with all fields", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+
+    const result = remandGenerator(pncUpdateDataset, operation)
+
+    expect(result).toStrictEqual({
+      operation: "NEWREM",
+      request: {
+        croNumber: null,
+        forceStationCode: "02YZ",
+        pncCheckName: null,
+        pncIdentifier: null,
+        arrestSummonsNumber: "11/01ZD/01/410780J",
+        hearingDate: "05122024",
+        nextHearingDate: "11122024",
+        pncRemandStatus: "B",
+        remandLocationCourt: "",
+        psaCourtCode: "9998",
+        courtNameType1: "Magistrates' Courts London Croydon MCA",
+        courtNameType2: "Magistrates' Courts London Croydon MCA",
+        localAuthorityCode: "0000",
+        bailConditions: ["This is a dummy bail condition."]
+      }
+    })
+  })
+
+  it("should set bail conditions to empty array when PNC remand status is 'C'", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+    pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.RemandStatus = "PB"
+
+    const result = remandGenerator(pncUpdateDataset, operation)
+
+    expect(result).toStrictEqual({
+      operation: "NEWREM",
+      request: {
+        croNumber: null,
+        forceStationCode: "02YZ",
+        pncCheckName: null,
+        pncIdentifier: null,
+        arrestSummonsNumber: "11/01ZD/01/410780J",
+        hearingDate: "05122024",
+        nextHearingDate: "11122024",
+        pncRemandStatus: "C",
+        remandLocationCourt: "",
+        psaCourtCode: "9998",
+        courtNameType1: "Magistrates' Courts London Croydon MCA",
+        courtNameType2: "Magistrates' Courts London Croydon MCA",
+        localAuthorityCode: "0000",
+        bailConditions: []
+      }
+    })
+  })
+
+  it("should set court house code to 1st instance warrant issued when there is a result qualifier 'LE' and result code is undated warrant", () => {
```

Comment:
Again, with these tests below, maybe we just test what changes?

Could also do something like:

```
expect(result).toStrictEqual({
  operation: "NEWREM",
  request: expect.objectContaining({
    courtNameType1: "*****1ST INSTANCE DATED WARRANT ISSUED*****",
    courtNameType2: "*****1ST INSTANCE DATED WARRANT ISSUED***** MCA"
  })
})
```